cli/cmd/inject.go

```diff
@@ -177,7 +179,20 @@ func injectObjectMeta(t *metaV1.ObjectMeta, k8sLabels map[string]string, options
  * injected, return false.
  */
 func injectPodSpec(t *v1.PodSpec, identity k8s.TLSIdentity, controlPlaneDNSNameOverride string, options *injectOptions, report *injectReport) bool {
-	// Pods with `hostNetwork=true` share a network namespace with the host. The
+	// check for ports with `protocol: udp`, which will not be routed by Linkerd
+	for _, container := range t.Containers {
```

Comment:
Could we get away with making this loop a function and then returning `true` as soon as we encounter a port with using UDP protocol. Something like:
```go

func checkUDPPorts(t *v1.PodSpec) bool{
	// check for ports with `protocol: udp`, which will not be routed by Linkerd
	for _, container := range t.Containers {
		for _, port := range container.Ports {
			if port.Protocol == v1.ProtocolUDP {
				return true
			}
		}
	}
	return false
}

func injectPodSpec(t *v1.PodSpec, identity k8s.TLSIdentity, controlPlaneDNSNameOverride string, options *injectOptions, report *injectReport) bool {
...
report.udp = checkUDPPorts(t)
```